Code/Framework/AzCore/AzCore/std/ranges/ranges_algorithm.h

```diff
@@ -1218,4 +1242,227 @@ namespace AZStd::ranges
+            constexpr auto operator()(R&& r, O result) const
+                -> enable_if_t<conjunction_v<
+                bool_constant<bidirectional_range<R>>,
+                bool_constant<bidirectional_iterator<O>>,
+                bool_constant<indirectly_copyable<iterator_t<R>, O>>
+                >, copy_backward_result<borrowed_iterator_t<R>, O>>
+            {
+                return operator()(AZStd::ranges::begin(r), AZStd::ranges::end(r), AZStd::move(result));
+            }
+        };
+    }
+    inline namespace customization_point_object
+    {
+        constexpr Internal::copy_fn copy{};
+        constexpr Internal::copy_if_fn copy_if{};
+        constexpr Internal::copy_n_fn copy_n{};
+        constexpr Internal::copy_backward_fn copy_backward{};
+    }
+
+    // ranges::move
+
+    // ranges::copy_backward
+    template<class I, class O>
+    using move_result = in_out_result<I, O>;
+    template<class I1, class I2>
+    using move_backward_result = in_out_result<I1, I2>;
+
+    namespace Internal
+    {
+        struct move_fn
+        {
+            template<class I, class S, class O>
+                constexpr auto operator()(I first, S last, O result) const
+                -> enable_if_t<conjunction_v<
+                bool_constant<input_iterator<I>>,
+                bool_constant<sentinel_for<S, I>>,
+                bool_constant<weakly_incrementable<O>>,
+                bool_constant<indirectly_movable<I, O>>
+                >, move_result<I, O>>
+            {
+                for (; first != last; ++first, ++result)
+                {
+                    *result = AZStd::ranges::iter_move(first);
+                }
+
+                return { AZStd::move(first), AZStd::move(result) };
+            }
+
+            template<class R, class O>
+            constexpr auto operator()(R&& r, O result) const
+                -> enable_if_t<conjunction_v<
+                bool_constant<input_range<R>>,
+                bool_constant<weakly_incrementable<O>>,
+                bool_constant<indirectly_movable<iterator_t<R>, O>>
+                >, move_result<borrowed_iterator_t<R>, O>>
+            {
+                return operator()(AZStd::ranges::begin(r), AZStd::ranges::end(r), AZStd::move(result));
+            }
+        };
+
+        struct move_backward_fn
+        {
+            template<class I1, class S1, class O>
+            constexpr auto operator()(I1 first, S1 last, O result) const
+                -> enable_if_t<conjunction_v<
+                bool_constant<bidirectional_iterator<I1>>,
+                bool_constant<sentinel_for<S1, I1>>,
+                bool_constant<bidirectional_iterator<O>>,
+                bool_constant<indirectly_movable<I1, O>>
+                >, move_backward_result<I1, O>>
+            {
+                for (I1 iter{ last }; iter != first;)
+                {
+                    *--result = AZStd::ranges::iter_move(--iter);
+                }
```

Comment:
question: Should this definitely be pre-decrement? 🤔 If we're pointing at the last element (and not `end`) then might be skip the last element? I could be getting things wrong I just wanted to check.

Reply:
I see, interesting, thanks for the link :+1: